frontend/app/services/MemberService.scala

```diff
@@ -1,136 +1,114 @@
+import com.gu.membership.salesforce.{ContactId, PaidTier}
 import com.gu.membership.stripe.Stripe.Customer
-import com.gu.membership.util.{FutureSupplier, Timing}
-import com.gu.membership.zuora.soap.Readers._
-import com.gu.membership.zuora.soap.actions.Actions.CreateFreeEventUsage
+import com.gu.membership.stripe.{Stripe, StripeService}
+import com.gu.membership.util.Timing
+import com.gu.membership.zuora.soap.models.PaymentSummary
 import com.gu.membership.zuora.soap.models.Queries.PreviewInvoiceItem
-import com.gu.membership.zuora.soap.models.Results.CreateResult
-import com.typesafe.scalalogging.LazyLogging
-import configuration.Config
+import com.gu.membership.zuora.soap.models.Results.{CreateResult, UpdateResult}
+import com.gu.services.PaymentService
 import controllers.IdentityRequest
 import forms.MemberForm._
-import model.Benefits.DiscountTicketTiers
-import model.Eventbrite.{EBCode, EBOrder, EBTicketClass}
-import model.FreeEventTickets
-import model.RichEvent._
-import monitoring.MemberMetrics
-import play.api.Play.current
-import play.api.libs.concurrent.Akka
-import play.api.libs.json.{JsObject, Json}
-import services.EventbriteService._
+import model.Eventbrite.{EBCode, EBTicketClass, EBOrder}
+import model.RichEvent.RichEvent
+import model._
+import org.joda.time.DateTime
+import play.api.Logger
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import services.api.ZuoraService.FeatureId
 import tracking._
+import views.support.ThankyouSummary
+import views.support.ThankyouSummary.NextPayment
 
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
-import scala.concurrent.duration._
 import scala.util.{Failure, Success}
 
 case class MemberServiceError(s: String) extends Throwable {
   override def getMessage: String = s
 }
 
-class FrontendMemberRepository(salesforceConfig: SalesforceConfig) extends ContactRepository {
-  val metrics = new MemberMetrics(salesforceConfig.envName)
+class MemberService(
+  identityService: => IdentityService,
+  salesforceService: => api.SalesforceService,
+  zuoraService: => api.ZuoraService,
+  stripeService: => StripeService,
+  catalogService: => api.CatalogService,
+  paymentService: => PaymentService) extends api.MemberService with ActivityTracking {
 
-  val salesforce = new Scalaforce {
-    val consumerKey = salesforceConfig.consumerKey
-    val consumerSecret = salesforceConfig.consumerSecret
 
-    val apiURL = salesforceConfig.apiURL.toString()
-    val apiUsername = salesforceConfig.apiUsername
-    val apiPassword = salesforceConfig.apiPassword
-    val apiToken = salesforceConfig.apiToken
+  import EventbriteService._
 
-    val stage = Config.stage
-    val application = "Frontend"
+  private val logger = Logger(getClass)
 
-    override val authSupplier: FutureSupplier[Authentication] = new FutureSupplier[Authentication](getAuthentication)
+  override def currentSubscription(contactId: ContactId): Future[model.Subscription] =
```

Comment:
it would be nice to reconcile this class with `SubscriptionService` in membership-common,
this currentSubscription method could probably call `findByProductFamily` in there

Given that this PR is dedicated to moving things rather than changing implementations though this comment is just a note for the future